packages/vue/test/components.test.ts

```diff
@@ -1,26 +1,120 @@
 import { formatComponentName } from '../src/components';
 
-it('properly format component names', () => {
-  const vm: any = {
-    $options: {},
-  };
-  vm.$root = vm;
-  expect(formatComponentName(vm)).toBe('<Root>');
-
-  vm.$root = null;
-  vm.$options.name = 'hello-there';
-  expect(formatComponentName(vm)).toBe('<HelloThere>');
-
-  vm.$options.name = null;
-  vm.$options._componentTag = 'foo-bar-1';
-  expect(formatComponentName(vm)).toBe('<FooBar1>');
-
-  vm.$options._componentTag = null;
-  vm.$options.__file = '/foo/bar/baz/SomeThing.vue';
-  expect(formatComponentName(vm)).toBe(`<SomeThing> at ${vm.$options.__file}`);
-  expect(formatComponentName(vm, false)).toBe('<SomeThing>');
-
-  vm.$options.__file = 'C:\\foo\\bar\\baz\\windows_file.vue';
-  expect(formatComponentName(vm)).toBe(`<WindowsFile> at ${vm.$options.__file}`);
-  expect(formatComponentName(vm, false)).toBe('<WindowsFile>');
+describe('formatComponentName', () => {
+  describe('when the vm is not defined', () => {
+    it('returns the anonymous component name', () => {
+      // arrange
+      const vm = undefined;
+
+      // act
+      const formattedName = formatComponentName(vm);
+
+      // assert
+      expect(formattedName).toEqual('<Anonymous>');
+    });
+  });
+
+  describe('when the vm is defined', () => {
+    let vm: any = undefined;
+
+    beforeEach(() => {
+      vm = {};
+    });
+
+    describe('and when the $options is not defined', () => {
+      it('returns the anonymous component name', () => {
+        // arrange
+        vm.$options = undefined;
+
+        // act
+        const formattedName = formatComponentName(vm);
+
+        // assert
+        expect(formattedName).toEqual('<Anonymous>');
+      });
+    });
```

Comment:
This is the main test I added.
Commenting the fix will make this test fail.
The rest is refactor (and the right-side ternary on `name` line 50 which was not covered).